### ISSUE TITLE
SharePoint2016 On Prem -SPFx HelloWorld App deployed to App Catalog  but not appearing under "Apps You Can Add"

### ISSUE THREAD

**Issue by theunitygamedev** [SharePoint/sp-dev-fx-webparts#331]
I have followed the SPFx tutorials to build the helloworld demo App and then uploaded to our On Prem SharePoint 2016 (with feature pack 2) app catalog.  All the properties show success as seen on screenshot below.  The problem is that the app is not visible on that site or subsite in the "Apps you can add" section.  I don't receive any on-screen errors either.  Any help will be greatly appreciated as SPFx seems like the way to go.

![helloworldappcatalog](https://user-images.githubusercontent.com/3999575/31521857-23768f88-af79-11e7-9b6a-ad7034af833c.png)

**Comment by Luksys5** [comment]
How did you build and packaged your solution?

**Comment by freddieje** [comment]
Hi Luksys5 and thanks for helping out:)

I followed the tutorial found here:

https://docs.microsoft.com/en-us/sharepoint/dev/spfx/web-parts/get-started/serve-your-web-part-in-a-sharepoint-page

To Package the solution I ran:
gulp package-solution

I then dragged the helloworld-webpart.sppkg file into our app catalog.  You can see the attached screenshots of the app in the catalog, but not it showing up when I try to add it to pages.  We got everything working in SharePoint online, but not on premise.

![helloworldappcatalogb](https://user-images.githubusercontent.com/3999575/31830140-56cc4dae-b58d-11e7-90f3-06a7fd0c4556.png)

![helloworldappnotshowingup](https://user-images.githubusercontent.com/3999575/31830164-618765ee-b58d-11e7-8207-033e7e4834d1.png)

**Comment by alinooshabadi** [comment]
Same problem here

**Comment by waldekmastykarz** [comment]
Which version of the SharePoint Framework generator are you using?

**Comment by alinooshabadi** [comment]
3.10.10

**Comment by waldekmastykarz** [comment]
Could it be that you're looking at a different version? The latest version of the SharePoint Framework available at the moment is v1.3.2. You can check the version number of the SharePoint Framework generator used to scaffold the project in the .yo-rc.json file in the root project folder. Then, in package.json, you can see the version of the (at)microsoft dependencies that are a part of the SharePoint Framework.

**Comment by alinooshabadi** [comment]
yes, my mistake. running 
`npm -v @microsoft/generator-sharepoint`
get yeoman version!

here is my package config

package.json
```json
{
  "dependencies": {
    "@microsoft/sp-core-library": "~1.1.0",
    "@microsoft/sp-webpart-base": "~1.1.0",
    "@microsoft/sp-lodash-subset": "~1.1.0"
  },
  "devDependencies": {
    "@microsoft/sp-build-web": "~1.1.0",
    "@microsoft/sp-module-interfaces": "~1.1.0",
    "@microsoft/sp-webpart-workbench": "~1.1.0"
  }
}
```

yo-rc.json
```json
{
  "@microsoft/generator-sharepoint": {
    "version": "1.3.2",
  }
}
```

**Comment by freddieje** [comment]
This is my package.json:

{
  "name": "hello-world",
  "version": "0.0.1",
  "private": true,
  "engines": {
    "node": ">=0.10.0"
  },
  "scripts": {
    "build": "gulp bundle",
    "clean": "gulp clean",
    "test": "gulp test"
  },
  "dependencies": {
    "@microsoft/sp-core-library": "~1.1.0",
    "@microsoft/sp-webpart-base": "~1.1.0",
    "@microsoft/sp-lodash-subset": "~1.1.0",
    "@types/webpack-env": ">=1.12.1 <1.14.0"
  },
  "devDependencies": {
    "@microsoft/sp-build-web": "~1.1.0",
    "@microsoft/sp-module-interfaces": "~1.1.0",
    "@microsoft/sp-webpart-workbench": "~1.1.0",
    "gulp": "~3.9.1",
    "@types/chai": ">=3.4.34 <3.6.0",
    "@types/mocha": ">=2.2.33 <2.6.0"
  }
}


yo-rc.json:

{
  "@microsoft/generator-sharepoint": {
    "version": "1.3.0",
    "libraryName": "hello-world",
    "libraryId": "b12fd61f-91c4-4589-97c9-e536820f199d",
    "environment": "onprem"
  }
}

**Comment by BasChrille** [comment]
Same problem here. Has anyone succeed in deploying to OnPrem?

**Comment by stluislu** [comment]
I also have the same problem!

**Comment by nazishqassim** [comment]
Hello,

Check your SP logs. You will see:

An exception was thrown while retrieving Apps from Corporate Catalog. Microsoft.SharePoint.SPException: One or more field types are not installed properly. Go to the list settings page to delete these fields. ---> System.Runtime.InteropServices.COMException: One or more field types are not installed properly. Go to the list settings page to delete these fields.

This issue is similar to https://github.com/SharePoint/sp-dev-docs/issues/1016

And it is evident from the initial screenshots in this issue that language dependent fix is not applied.
Read https://blogs.technet.microsoft.com/stefan_gossner/2017/11/14/november-2017-cu-for-sharepoint-server-2016-is-available-for-download/

If you see the last column in the first screenshot of this issue, it is showing the resource labels. Actually the column name is "Deployed".

In short, install the language dependent update that comes with the CU, then delete the App Catalog, create the catalog again, upload the app, give rights on the app catalog to the user. And then finally you will see your app in "Add an app" section.   

Bye

**Comment by panjkov** [comment]
I have similar issue, everything is set properly and column name is visible in the app catalog. 
I am building webpart using generator v1.4. 
I can add an app to site, but it is not available when I want to add the webpart to page from ribbon.

**Comment by nazishqassim** [comment]
Hi, I tried with generator 1.4 and web part did appear properly. Assuming you have selected the correct option when generating from yeoman "SharePoint 2016 onwards..including SharePoint Online".
Besides, the web part will appear in the "Other" rather than "Custom" web part category. Thanks

**Comment by panjkov** [comment]
Yes I have everything set up properly.
I reported my issue here https://github.com/SharePoint/sp-dev-docs/issues/1188
________________________________

**Comment by beinguam** [comment]
nazishqassim nailed it - was the exact solution for me

**Comment by hjaouani** [comment]
I have the same problem.
Is there any solution or suggestion ?

**Comment by nazishqassim** [comment]
You can either follow the message above in which I explained about the actual error appearing in SP log. In that case, install language dependent cu and recreate app catalog. If that is not the issue then look at panjkov issue. Thanks

**Comment by lifayt** [comment]
I'm experiencing the same problem - the addendum is that I'm not seeing that log error, and I have the language dependent fix installed. Any clues what could be causing this?

**Comment by Pieter-Veenstra** [comment]
I just had the same problems on 2 out of 6 farms. I had to recreate the app catalog site collection and all is now working.

**Comment by sherbertas** [comment]
I had a similar issue - App seems to be deployed in App Catalog site, but not available in other sites. I noticed, that even App is deployed in App Catalog, it still has a ribbon button "Deploy" enabled in the list. I clicked the Deploy button once again, and then everything worked as expected.

**Comment by smoothsen** [comment]
+1

**Comment by JayakumarB** [comment]
@sherbertas please confirm whether you got deploy window with the checkbox which is highlighted below,
![image](https://user-images.githubusercontent.com/11646986/41228165-0636e910-6d95-11e8-8817-930ed95e2c0b.png)


If you click on the checkbox the app will be deployed to all the sites in the tenant and you no need it install the app specifically you can start adding the webparts in the pages directly.

**Comment by marcelaopimenta** [comment]
I had this problem. I update my SharePoint environment with lastest cumulative update (including Language Packs) and done!

**Comment by vmannenvc** [comment]
I followed all the options and workarounds above yet not seeing the client side webpart. @marcelaopimenta  can you please share link to cu,cumulative updates?

**Comment by marcelaopimenta** [comment]
Here have all cumulatives update... Install the last.. JUN 2018

https://www.toddklindt.com/blog/Builds/SharePoint-2016-Builds.aspx

**Comment by vmannenvc** [comment]
Thanks, was able to resolve the issue with latest updates. Additionally had to start services as outlined here to make it work. http://sharepointchick.com/archive/2012/07/29/setting-up-your-app-domain-for-sharepoint-2013.aspx

**Comment by rajkrishnanet** [comment]
Hi All ,I am facing on-premise deployment issue. I have installed April 2018 feature pack(KB4018340 and related language packs) and created App catalog. The Client webpart deployed successfully in On Premise Appcatalog but webpart is not available in "Add an App" page.I am maintaining CDN artifacts in roo site collection root site.
 Please help me on this issue. 
vmannenvc - what are  the services need to start?

**Comment by katika666** [comment]
Hi all, 
My installation environment is SharePoint19 and, 
npm -v @microsoft/generator-sharepoint 
6.5.0
As I see I steel need downgrade framework for correct use of web Parts especially: Modern Script Editor.

**Comment by kbkotresh** [comment]
@VesaJuvonen, 
My Current Sharepoint version is SP2016 with DEC 2018 CU updated . I have developed the Sample webpart and Details in (https://github.com/SharePoint/sp-dev-docs/issues/3200)

I am also facing same issue not able to see the SPFX webpart in "Add App" SP2016 environment. 

Kindly suggest

**Comment by kbkotresh** [comment]
> @VesaJuvonen,
> My Current Sharepoint version is SP2016 with DEC 2018 CU updated . I have developed the Sample webpart and Details in ([SharePoint/sp-dev-docs#3200](https://github.com/SharePoint/sp-dev-docs/issues/3200))
> 
> I am also facing same issue not able to see the SPFX webpart in "Add App" SP2016 environment.
> 
> Kindly suggest

I am able to resolve this issue .
1. Apply Language Dependent Files October 2018 CU Updates. 
2.Run  Psconfig.exe 
3. Delete the Existing App catalog and recreate new App catalog. 
4. Add app to app catalog 

Thanks for supports
Best regards,
Kotresh

**Comment by ymihir** [comment]
I have created a web part by selecting SP2019 and SPOnline as option. Web part is working fine in SP Online but same package is not working in SP 2019. SP 2019 is fresh installation and i haven't done any CU updates. This could be the problem? Also I have made `includeClientSideAssets:true` .

**Comment by katika666** [comment]
> I have created a web part by selecting SP2019 and SPOnline as option. Web part is working fine in SP Online but same package is not working in SP 2019. SP 2019 is fresh installation and i haven't done any CU updates. This could be the problem? Also I have made `includeClientSideAssets:true` .

For SP2019 on Prem, you need to downgrade framework during deployment. download old version from repository.

**Comment by vasavidurgaprasad** [comment]
Hi All,
I am also facing same issue in SP2016 dev farm. The webpart is not visible after deploying.
I have installed July 2019 CU patch. The Yomen generated template version i am using is 1.8.2 and Node Version is 10.16.2. 
Do I need to downgrade both Node and Yomen template?

**Comment by emadhejazian** [comment]
I have the same issue, anyone has a solution?

**Comment by emadhejazian** [comment]
https://support.shortpoint.com/support/solutions/articles/1000273552
